### PR TITLE
Bump to 6.0.0 with fixes

### PR DESCRIPTION
**Pull request by traversaro**
This PR implements part 2 of the plan described in https://github.com/conda-forge/msgpack-c-feedstock/pull/39#issuecomment-1378387632 . In particular, switch to use `c-<version>` release tags.

Fix https://github.com/conda-forge/msgpack-c-feedstock/pull/30
Fix https://github.com/conda-forge/msgpack-c-feedstock/pull/40
Fix https://github.com/conda-forge/msgpack-c-feedstock/pull/41

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
The PR is ready for review, even if we may want to merge it after https://github.com/conda-forge/staged-recipes/pull/22365 to make sure that we still have a msgpack-cxx package available.

**Comment by hmaarrfk** [comment]
This https://github.com/conda-forge/staged-recipes/pull/22365 has been merge

Lets merge this PR in a bit.

**Comment by traversaro** [comment]
@conda-forge/msgpack-c does anyone has additional feedback on this PR? Thanks!

**Comment by ndevenish** [comment]
So, anyone using this package for C++ will break?

**Comment by hmaarrfk** [comment]
Previous builds should be fine since a "run_export" is used. So old environments will still resolve correctly.

There will be some churn from maintainers coming to upgrade `mspack-c` if they are using cpp, they will have to add/change to `msgpack-cxx`. 

This churn seems acceptable.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,25 +28,20 @@ requirements:
...
-    - boost-cpp
 
 test:
   commands:
     - test -f $PREFIX/include/msgpack.h                         # [unix]
-    - test -f $PREFIX/include/msgpack.hpp                       # [unix]
-    - test -f $PREFIX/lib/libmsgpackc.a                         # [unix]
-    - test -f $PREFIX/lib/libmsgpackc.so                        # [linux]
-    - test -f $PREFIX/lib/libmsgpackc.dylib                     # [osx]
+    - test -f $PREFIX/lib/libmsgpack-c.a                        # [unix]
```
Comment (hmaarrfk) [inline recipe/meta.yaml]:
```suggestion
    - test ! -f $PREFIX/lib/libmsgpack-c.a                        # [unix]
```

File: recipe/build.sh [hunk 2]
```diff
@@ -8,7 +8,6 @@ cmake ${CMAKE_ARGS} \
     -GNinja \
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (hmaarrfk) [inline recipe/build.sh:10]:
```suggestion
    -DCMAKE_INSTALL_PREFIX=$PREFIX \
    -DBUILD_SHARED_LIBS=ON \
```

File: recipe/build.sh [hunk 3]
```diff
@@ -8,7 +8,6 @@ cmake ${CMAKE_ARGS} \
...
-    -DMSGPACK_CXX11=YES \
```
Comment (hmaarrfk) [inline recipe/build.sh:11]: looking at their cmake file, this seems to disable the static builds, helping with the test to assert lack of static binaries.

**Comment by hmaarrfk** [comment]
@traversaro  I mostly came to check if you were going to add yourself as a maintainer, but found two small nits. 


Would you like to be a maintainer?

File: recipe/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipe/meta.yaml]: Ok, that make sense as per [CFEP-018](https://github.com/conda-forge/cfep/blob/main/cfep-18.md). However, probably I can do the same for Windows for consistency?

**Comment by traversaro** [comment]
> 
> Would you like to be a maintainer?

Sure, I added myself as part of the PR.

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml]: I'm just not as adept with windows. So I am not able to provide stronger recommendations there. But yes, doing the same for windows would be preferred.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
rerendering adds you to the CODEOWNERS fyi

**Pull request merged**